### PR TITLE
compress.gzip / deflate: rename gzip module into deflate module

### PR DESCRIPTION
**Pull request by ChAoSUnItY**
This change is suggested by @Le0Developer from [#14559's comment](https://github.com/vlang/v/pull/14599#issuecomment-1146387340)
- Rename `compress.gzip` into `compress.deflate`.
- Modify test `deflate_test.v` (Originally `gzip_test.v`)
```v
module deflate

const gzip_magic_numbers = [u8(0x1f), 0x8b]

fn test_gzip() ? {
	uncompressed := 'Hello world!'
	compressed := compress(uncompressed.bytes())?
	first2 := compressed[0..2]
	assert first2 != deflate.gzip_magic_numbers
	decompressed := decompress(compressed)?
	assert decompressed == uncompressed.bytes()
}
```

### CODE REVIEW

File: vlib/compress/deflate/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+## Description:
+
+`compress.deflate` is a module that assists in the compression and
+decompression of binary data using `deflate` compression
+
+NOTE: To decompress gzip, discard first 10 bytes of compressed bytes
+then use `compress.deflate.decompress`. (Header validation won't be
+performed in this case)
```
Comment (Le0Developer) [inline vlib/compress/deflate/README.md:8]: This is wrong as gzip headers *can* be longer and gzip also has an 8 byte checksum at the end.

**Review by Le0Developer — COMMENTED** [review]
Good work! Suddenly got the inspiration to actually implement gzip, so I will update the comment in the following PR.

File: vlib/compress/deflate/README.md (hunk 1 above)
Comment (ChAoSUnItY) [reply vlib/compress/deflate/README.md:8]: maybe you can rewrite the description when your gzip implementation is done? I would be appreciate for this.

Comment (Le0Developer) [reply vlib/compress/deflate/README.md:8]: I will! Currently writing tests for it and should be ready soon.

**Pull request merged**